libs/blocks/preflight/panels/assets.js

```diff
@@ -0,0 +1,188 @@
+import { html, signal, useEffect } from '../../../deps/htm-preact.js';
+import { createTag } from '../../../utils/utils.js';
+
+function isViewportTooSmall() {
+  return !window.matchMedia('(min-width: 1200px)').matches;
+}
+
+const imagesWithMismatch = signal([]);
+const imagesWithMatch = signal([]);
+const checksPerformed = signal(false);
+const viewportTooSmall = signal(isViewportTooSmall());
+
+const groups = [
+  { title: 'Images with dimension mismatch', imgArray: imagesWithMismatch },
+  { title: 'Images with matching dimensions', imgArray: imagesWithMatch },
+];
+
+async function checkImageDimensions() {
+  if (checksPerformed.value || viewportTooSmall.value) return;
+  checksPerformed.value = true;
+
+  const allImages = [...document.querySelectorAll('main picture img')];
+  if (!allImages.length) return;
+
+  // Force load all images
+  await Promise.all(
```

Comment:
If any image fails to load, the Promise.all() will reject and the entire image checking process will stop. It probably would make sense to catch this type of case.

Reply:
True! Instead of a `catch`, I'll add an error listener for the image itself, should be more straightforward.